modules/jvm/bash/artifacts/opt/jboss/container/java/jvm/java-default-options

```diff
@@ -160,4 +166,4 @@ error_handling() {
 initialize_container_limits > /dev/null
 
 ## Echo options, trimming trailing and multiple spaces
-echo "$(initial_memory) $(max_memory) $(gc_config) $(diagnostics) $(error_handling)" | awk '$1=$1'
+echo "$(initial_memory) $(max_memory) $(max_ram_percentage) $(gc_config) $(diagnostics) $(error_handling)" | awk '$1=$1'
```

Comment:
What are the defaults for `initial_memory` and `max_memory`? Would those set `-Xms` and -Xmx` values by default (i.e. if nothing has been specified by the user? If so, setting `-XX:MaxRAMPercentage` doesn't make sense as explicit `-Xmx` settings take precedence.

Reply:
Ugh. I haven't been able to quickly confirm what circumstances will result in -Xmx and -Xms being set by the launch scripts. My position is: if the user effectively specifies those directly, then we should not detect/prevent that, just as they could do so by running `java` themselves: let the runtime decide what to do. But yes, if there's an indirect way we cause `-Xmx` or `-Xms` to be specified we should ensure that this makes sense wrt to this new option.

I think the circumstance where that might happen is you have a container runtime imposed memory limit on a cgroups v1 system. I need to make some adjustments to my setup to test that scenario properly yet.

(I am looking forward to dumping all this stuff in 9)